### PR TITLE
add image refs, env vars, so we can get jenkins images from payload

### PR DESCRIPTION
**Pull request by gabemontero**
/assign @bparees 

ptal to review at least the yaml/ref stuff

in theory we could just merge this, but ideally I can wire in the processing of the new env's in the samples operator code with this PR as well

I'd like to start on those after https://github.com/openshift/cluster-samples-operator/pull/81

### CODE REVIEW

**Comment by bparees** [comment]
@gabemontero it looks correct to me.

**Comment by gabemontero** [comment]
ah  ... I need to wait until the mirroring for jenkins-agent-base completes:

`error: unable to create a release: operator "cluster-samples-operator" failed to map images: image file "/tmp/release-image-0.0.1-2019-01-23-165028825421902/cluster-samples-operator/image-references" referenced image "jenkins-agent-base" that is not part of the input images`

**Comment by gabemontero** [comment]
updated for https://github.com/openshift/release/pull/2678 @bparees

File: deploy/05-operator.yaml [hunk 1]
```diff
@@ -36,4 +36,12 @@ spec:
...
             - name: OPERATOR_NAME
-              value: "cluster-samples-operator"
+              value: cluster-samples-operator
+            - name: IMAGE_JENKINS
+              value: quay.io/openshift/origin-jenkins:latest
+            - name: IMAGE_AGENT_BASE
+              value: quay.io/openshift/origin-jenkins-agent-base:latest
+            - name: IMAGE_NODEJS_AGENT
+              value: quay.io/openshift/origin-jenkins-agent-nodejs:latest
+            - name: IMAGE_MAVEN_AGENT
```
Comment (bparees) [inline deploy/05-operator.yaml]: might as well fix the env var names too.

Comment (gabemontero) [reply deploy/05-operator.yaml]: fixed

File: deploy/image-references [hunk 2]
```diff
@@ -6,3 +6,19 @@ spec:
...
       kind: DockerImage
       name: docker.io/openshift/origin-cluster-samples-operator:latest
+  - name: jenkins
+    from:
+      kind: DockerImage
+      name: quay.io/openshift/origin-jenkins:latest
+  - name: jenkins-agent-base
+    from:
+      kind: DockerImage
+      name: quay.io/openshift/origin-jenkins-agent-base:latest
```
Comment (bparees) [inline deploy/image-references]: i don't think we actually need or want the jenkins-agent-base reference.   right now i'm thinking we will not have an imagestream for it.

Comment (gabemontero) [reply deploy/image-references]: agent-base removed

**Comment by bparees** [comment]
image refs/envs look ok to me again

**Comment by bparees** [comment]
@gabemontero i'd be ok merging this now so we can start seeing if the env vars show up properly on the samples operator.  May make it easier for you to iterate on the next steps later when the cluster is populating the payload correctly out of the box.

**Comment by bparees** [comment]
but i think we are supposed to wait on this until the OCP side of the payload is ready which it is not yet.

/hold

**Comment by bparees** [comment]
discussion of the next step to get it into the OCP payload:
https://coreos.slack.com/archives/CB95J6R4N/p1548279126559100

**Comment by bparees** [comment]
I do not see the jenkins images in the ocp imagestream, so you'll have to bug @joeldavis84 or @tbielawa to see what we've missed in terms of getting the new jenkins images built+published for ocp.

**Comment by gabemontero** [comment]
@bparees yeah, I see those ist's now in the ocp namespace (appear to have been there since Jan 24:

```
gmontero ~/go/src/github.com/openshift/release  (master)$ oc get imagestreamtags -n ocp | grep jenkins
4.0-art-latest:jenkins-agent-maven                                               quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:3e3d25d299cb437862d55266ba5ff782e101fe46479ae2b0664965f20aca50c4                          2 days ago
4.0-art-latest:jenkins-agent-nodejs                                              quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:c76ed9d6fd59e9b6e4c6e9a71d87bb10a29562a14a453c25fe34cea46e0eb9cd                          2 days ago
4.0-art-latest:jenkins-agent-base                                                quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:702bf0228ff662cd43ce26f7ee35f4b7ecb30aaf4e8b95c46b52a300701c1e48                          2 days ago
4.0-art-latest:jenkins                                                           quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:2ca6e2558ebe419ed61a264f2c99f51015b2d041acce906c867dce538f0fcdd6                          2 days ago
4.0-art-latest:jenkins-2-rhel7                                                   quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:6dc87ecd037515989f9620efbdb27254e3ea358c405b909d1335b240538b3e63                          4 days ago

```

I would think the naming issue I noted above is the smoking guy currently.

**Comment by bparees** [comment]
it's not just that they need to be in the ocp namespace, they need to be part of the "4.0" imagestream.

` oc get is 4.0 -n ocp -o yaml `

if you look in that imagestream (named "4.0") you will see tags for all the other OCP payload images.

It is supposed to be the equivalent to the "origin-v4.0" imagestream in the "openshift" namespace:

```
oc get is origin-v4.0 -n openshift -o yaml | grep -i jenkins
    name: jenkins
    name: jenkins-agent-base
    name: jenkins-agent-maven
    name: jenkins-agent-nodejs
    tag: jenkins
    tag: jenkins-agent-base
    tag: jenkins-agent-maven
    tag: jenkins-agent-nodejs
```

**Comment by gabemontero** [comment]
ok @bparees showed me what I was overlooking/blind to in my image ref file ... pushed update ... will confirm if we at least get past that and get to what @bparees has been noting re: the ocp image stream

**Comment by gabemontero** [comment]
In the failed image eco run, the `machine-config-operator` did not reach availability in time ; I saw http level error messages in its pod logs

**Comment by gabemontero** [comment]
from @bparees on slack:  things look good re: ocp ... needed to look at `oc get is 4.0-art-latest -n ocp -o yaml`

**Comment by gabemontero** [comment]
@bparees this still needs to hold until the other PRs merge, but this is reviewable in that

a) the changes do not collide with https://github.com/openshift/cluster-samples-operator/pull/81
b) currently assumes that the OKD choice is "going away" :-) ... I could add qualifiers to only tag in the payload with OCP if we want ... but for now I have it up as the simpler change

**Comment by gabemontero** [comment]
you convinced me @bparees  .... placed on the merge queue

**Pull request merged**